src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -615,87 +616,17 @@ CitusCreateDirectory(StringInfo directoryName)
 
 
 /*
- * CitusRemoveDirectory first checks if the given directory exists. If it does, the
- * function recursively deletes the contents of the given directory, and then
- * deletes the directory itself. This function is modeled on the Boost file
- * system library's remove_all() method.
+ * CitusRemoveDirectory removes the given directory, and errors out if
+ * there were any problems.
  */
 void
 CitusRemoveDirectory(StringInfo filename)
 {
-	struct stat fileStat;
-	int removed = 0;
-
-	int fileStated = stat(filename->data, &fileStat);
-	if (fileStated < 0)
-	{
-		if (errno == ENOENT)
-		{
-			return;  /* if file does not exist, return */
```

Comment:
This simply returns, [but `rmtree` throws an error](https://github.com/postgres/postgres/blob/master/src/common/pgfnames.c#L43). This should cause at least one test to fail. And if it doesn't we should add a test which exercises this path!